drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -676,7 +676,18 @@ struct bcm2835_isp_fmt *get_default_format(struct bcm2835_isp_node *node)
 static inline unsigned int get_bytesperline(int width,
 					    const struct bcm2835_isp_fmt *fmt)
 {
-	return ALIGN((width * fmt->depth) >> 3, fmt->bytesperline_align);
+	unsigned int bpl = (width * fmt->depth) >> 3;
+	int align = fmt->bytesperline_align;
+	int align_minus_1 = align - 1;
+	if (align & align_minus_1) {
+		/* align is not a power of 2 */
+		unsigned int remainder = bpl % align;
+		if (remainder)
+			bpl += (align - remainder);
+	}
+	else
+		bpl = (bpl + align_minus_1) & ~align_minus_1;
+	return bpl;
 }
```

Comment:
Not entirely sure, but I would guess the upstream folks might object to using align_minus_1 variable over an explicit (align - 1).

Reply:
No problem. Given that I was using  it several times I thought I'd make it explicit that I expected the value to be re-used, but if we think (a) the compiler would do it anyway and (b) upstream maintainers might object, then I'll happily change it. Is that the general feeling?

Reply:
They would probably prefer you to use the existing ALIGN and IS_ALIGNED macros.

Reply:
Personally I don't think we need the complication with manually working out if it's a bitmask or not.
24bpp is the only format that has quirks, and you have a near match with https://github.com/raspberrypi/linux/blob/rpi-5.4.y/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c#L994
```
if (fmt->depth == 24)
   return <whatever the correct align is here, and it doesn't need to be 96>
else
  return ((width * fmt->depth) >> 3, fmt->bytesperline_align);
```
Admittedly it's not a performance critical path, but manually computing whether we need to do the modulo seems nuts.

MMAL sets the stride via a multiple of width * bpp. 32pixels * 24bpp is 96 bytes, which fulfils the ISP requirement for a multiple of 32 bytes. So I think we can do 
```
  return ((width, fmt->bytesperline_align) * fmt->depth) >> 3);
```
and leave the bytesperline_align at 32 (or hardcode it here).